src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,270 @@
+        public virtual StackItem Transfer(ApplicationEngine engine, Array args)
+        {
+            UInt160 from = new UInt160(args[0].GetSpan());
+            UInt160 to = new UInt160(args[1].GetSpan());
+            BigInteger amount = args[2].GetBigInteger();
+            byte[] tokenId = args[3].GetSpan().ToArray();
+            return Transfer(engine, from, to, amount, tokenId);
+        }
+
+        public virtual bool Transfer(ApplicationEngine engine, UInt160 from, UInt160 to, BigInteger amount, byte[] tokenId)
+        {
+            if (amount.Sign < 0) throw new ArgumentOutOfRangeException(nameof(amount));
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, from)) return false;
+
+            var storages = engine.Snapshot.Storages;
+            ContractState contract_to = engine.Snapshot.Contracts.TryGet(to);
+            if (contract_to?.Payable == false) return false;
+            if (!amount.IsZero && !from.Equals(to))
+            {
+                UInt160 innerKey = GetInnerKey(tokenId);
+                StorageKey fromKey = CreateOwnershipKey(from, innerKey);
+                UState fromBalance = storages.GetAndChange(fromKey)?.GetInteroperable<UState>();
+                if (fromBalance is null) return false;
+                if (fromBalance.Balance < amount) return false;
+                if (fromBalance.Balance == amount)
+                {
+                    storages.Delete(fromKey);
+                    storages.Delete(CreateOwnershipKey(innerKey, from));
+                }
+                fromBalance.Balance -= amount;
+
+                StorageKey toKey = CreateOwnershipKey(to, innerKey);
+                UState toBalance = storages.GetAndChange(toKey, () => new StorageItem(new UState())).GetInteroperable<UState>();
+                toBalance.Balance += amount;
+                storages.GetAndChange(CreateOwnershipKey(innerKey, to), () => new StorageItem(new byte[0]));
+            }
+            engine.SendNotification(Hash, new Array(new StackItem[] { "Transfer", from.ToArray(), to.ToArray(), amount, tokenId }));
+            return true;
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.ByteArray }, ParameterNames = new[] { "tokenId" })]
+        protected StackItem Properties(ApplicationEngine engine, Array args)
+        {
+            byte[] tokenid = args[0].GetSpan().ToArray();
+            return Properties(engine.Snapshot, tokenid).ToString();
+        }
+
+        public abstract JObject Properties(StoreView snapshot, byte[] tokenid);
+
+        internal protected virtual void Mint(ApplicationEngine engine, UInt160 account, byte[] tokenId)
```

Comment:
Should we be able to `Mint` to <code>enumerator</code> that contains possible co-owners?
In that case we would need to specify the amount of each co-owner.

Reply:
> There are some possible scenarios that are useful, without that we may need a third party to share the ownership.

They can use `Transfer` method, which support co-owners if `Decimals` > 0.

Reply:
But who will be the first owner? Can this `transfer` directly happen?

Reply:
I think it should be decided by administrator, as `Mint` is not a standard method of Nep11, it's only for admin.

Reply:
Ok, @Tommo-L, I agree.